tests/S-4_Account_creation _deletion_and _get_info.test.js

```diff
@@ -0,0 +1,369 @@
+const http = require('http');
+const listen = require('test-listen');
+const got = require('got');
+
+// Initialize server
+test.before(async (t) => {
+    console.log("Creating test server")
+    t.context.server = http.createServer(app);
+    t.context.prefixUrl = await listen(t.context.server);
+    t.context.got = got.extend({ prefixUrl: t.context.prefixUrl, responseType: 'json' });
+    console.log("Successfully created test server")
+})
+
+// Shutdown server
+test.after.always(async (t) => {
+    console.log("Testing finished, closing test server")
+    await t.context.server.close();
+    console.log("Successfully closed test server")
+
+})
+
+
+
+// =========================== POST /admin/user =========================== //
+
+// Test for:    POST    /admin/user
+test("POST endpoint /admin/user", async (t) => {
+
+    const { statusCode } = await t.context.got.post("admin/user", {
+        // Give it dummy data 
+        json: {
+            surname: "surname_dummy",
+            name: "name_dummy",
+            id: 0,
+            userName: "userName_dummy",
+            email: "email@dummy_data"
+        }
+    });
+
+    // Check status Code 
+    t.is(statusCode, 200, "Expected status code 200 for successful post");
+
+});
+
+// Test for Multipule:    POST    /admin/user
+test("Multipule POST endpoint /admin/user", async (t) => {
+
+    const usersToCreate = [
+        {
+            surname: "surname_dummy1",
+            name: "name_dummy1",
+            id: 1,
+            userName: "userName_dummy1",
+            email: "email1@dummy_data"
+        },
+        {
+            surname: "surname_dummy2",
+            name: "name_dummy2",
+            id: 2,
+            userName: "userName_dummy2",
+            email: "email2@dummy_data"
+        },
+        {
+            surname: "surname_dummy3",
+            name: "name_dummy3",
+            id: 3,
+            userName: "userName_dummy3",
+            email: "email3@dummy_data"
+        },
+    ];
+
+    for (const userData of usersToCreate) {
+        const { statusCode } = await t.context.got.post("admin/user", {json: userData});
+
+        // Check status Code 
+        t.is(statusCode, 200, `Expected status code 200 for successful deletion, Bad request for user ${userData.userName}`);
+       
+    }
+});
+
+// It should be a bit different but what can we do at this point
+// Test for Wrong input:    POST    /admin/user
+test("Wrong input POST endpoint /admin/user", async (t) => {
+
+    const { statusCode } = await t.context.got.post("admin/user", { 
+        // Give it dummy data 
+        json: {
+        //     surname: "surname_dummy",
+        //     name: "name_dummy",
+        //     id: 0,
+        //     userName: "userName_dummy",
+        //     email: "email@dummy_data"
+        }
+    });
+
+    // Check status Code 
+    // It shouldn't respond with 200 when the body is empty, So now the correct step would be to change the function
```

Comment:
Or to accept life as it is (we will leave such comments in the code though, just to remind the teacher that all this doesn't make any sense)